modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -33,78 +47,108 @@ UnitTest.asynctest('browser.tinymce.plugins.fullscreen.FullScreenPluginTest', (s
+          })
+        ]),
+        Chain.fromIsolatedChains([
+          Chain.mapper(getContentContainer),
+          UiFinder.cFindIn('.tox-silver-sink.tox-tinymce-aux'),
+          Chain.op((sink) => {
+            Assertions.assertEq('Editor sink z-index', shouldExist ? '1201' : '1300', Css.get(sink, 'z-index'));
+          })
+        ])
+      ])
+    );
+
+  const cAssertShadowHostState = (label: string, shouldExist: boolean) =>
+    Chain.label(
+      `${label}: Shadow host should ${shouldExist ? '' : 'not'} have "tox-fullscreen" and "tox-shadowhost" classes and z-index`,
+      Chain.op((editor: Editor) => {
+        if (SugarShadowDom.isInShadowRoot(SugarElement.fromDom(editor.getElement()))) {
+          const host = SugarShadowDom.getShadowRoot(SugarElement.fromDom(editor.getElement()))
+            .map(SugarShadowDom.getShadowHost)
+            .getOrDie('Expected shadow host');
 
-  const cSetupEditor = McEditor.cFromSettings({
+          Assertions.assertEq('Shadow host classes', shouldExist, Classes.hasAll(host, [ 'tox-fullscreen', 'tox-shadowhost' ]));
+          Assertions.assertEq('Shadow host z-index', shouldExist ? '1200' : 'auto', Css.get(host, 'z-index'));
+        }
+      }));
+
+  const cAssertPageState = (label: string, shouldExist: boolean) =>
+    Chain.fromChains([
+      cAssertHtmlAndBodyState(label, shouldExist),
+      cAsssertEditorContainerAndSinkState(label, shouldExist),
+      cAssertShadowHostState(label, shouldExist)
+    ]);
+
+  TinyLoader.setupInBodyAndShadowRoot((editor: Editor, onSuccess, onFailure) => {
+    Pipeline.async({}, [
+      Log.chainsAsStep('TBA', 'FullScreen: Toggle fullscreen on, open link dialog, insert link, close dialog and toggle fullscreen off', [
+        Chain.inject(editor),
+        Chain.fromParent(Chain.identity, [
+          cAssertPageState('Before fullscreen command', false),
+          Chain.op((editor: Editor) => editor.execCommand('mceFullScreen', true)),
+          cAssertApiAndLastEvent('After fullscreen command', true),
+          cAssertPageState('After fullscreen command', true),
+          Chain.op((editor: Editor) => editor.execCommand('mceLink', true)),
+          cWaitForDialog('Insert/Edit Link'),
+          cCloseOnlyWindow,
+          cAssertPageState('After window is closed', true),
+          Chain.op((editor: Editor) => editor.execCommand('mceFullScreen')),
+          cAssertApiAndLastEvent('After fullscreen toggled', false),
+          cAssertPageState('After fullscreen toggled', false)
+        ])
+      ]),
+      Log.chainsAsStep('TBA', 'FullScreen: Toggle fullscreen and remove editor should clean up classes', [
+        Chain.inject(editor),
+        Chain.fromParent(Chain.identity, [
+          Chain.op((editor: Editor) => editor.execCommand('mceFullScreen', true)),
+          cAssertApiAndLastEvent('After fullscreen command', true),
+          cAssertPageState('After fullscreen command', true)
+        ]),
+        McEditor.cRemove,
```

Comment:
This shouldn't be here, as you've not constructed an editor using `McEditor`. If you're switching to using TinyLoader, then we'll need to find another way to do this.

Reply:
Looks like it just wasn't deleted when changing the tests?

Reply:
I have restructured the code so this can be kept to allow the class removal logic on editor removal to be tested.